components/global/Header/Subnavigation.js

```diff
@@ -1,34 +1,75 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
+/*
+ * Ignore false positives (key event exists on Navigation component; role attribute not needed
+ * since Link component validly handles href and window history, click just closes subnav)
+ */
+import { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import Link from "next/link";
 import classNames from "classnames";
 import internalLinkShape from "@/shapes/link";
+import NavItemWithChildren from "./NavItemWithChildren";
+
+export default function Subnavigation({
+  items,
+  active,
+  onClick,
+  theme,
+  baseClassName = "c-subnav-list",
+  level = 2,
+}) {
+  const [activeSub, setActiveSub] = useState(null);
+
+  useEffect(() => {
```

Comment:
does this need an effect? I see that it's there to prevent endless renders setting `activeSub` after `active` changes, but it looks like it could be removed, the only replacement necessary is in the render doing:

```
        const isActiveSub = active && id === activeSub;
```

otherwise we're splitting the state into two separate updates, which can feel sluggish

Reply:
good catch.  The management of like open state was something I circled back on after the fact, and you're right, in this case it is more practical to just use the value of the `active` prop rather than use it to trigger internal state change.

Reply:
Ooo.  Actually, I take it back.  I think this is the best (only?) way I know how to ensure 3rd level accordions are collapsed if their parent dropdown is closed.  Which I'm doing so that navs start out all neat and collapsed every time their parent dropdown is opened.